src/utils/network-monitor.ts

```diff
@@ -1570,47 +1577,54 @@ export class NetworkMonitor {
     canFail = false,
     interval = 5000,
   }: BlockParams): Promise<ExtendedBlock | null> {
-    return new Promise<ExtendedBlock | null>((topResolve, _topReject) => {
-      let counter = 0
-      let sent = false
-      let blockInterval: NodeJS.Timeout | null = null
-      const getBlock = async (): Promise<void> => {
+    let counter = 0
+    let sent = false
+
+    const getBlock = async (): Promise<ExtendedBlock | null> => {
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
         try {
           const block: ExtendedBlock | null = await getExtendedBlock(this.providers[network], blockNumber)
           if (block === null) {
             counter++
             if (canFail && counter > attempts) {
-              if (blockInterval) clearInterval(blockInterval)
               if (!sent) {
                 sent = true
-                topResolve(null)
               }
+
+              return null
             }
           } else {
-            if (blockInterval) clearInterval(blockInterval)
             if (!sent) {
               sent = true
-              topResolve(block as ExtendedBlock)
             }
+
+            return block as ExtendedBlock
           }
         } catch (error: any) {
           if (error.message !== 'cannot query unfinalized data') {
```

Comment:
Similar to my previous message. I wonder if we dont up the counter on any error we may get stuck in a loop.

Reply:
iirc I think this is a common error returned from ethers when the data is just not ready yet so maybe that's why this particular error is skipped. Not 100% sure